lib/ConnectionConfig.js

```diff
@@ -51,32 +45,42 @@ function ConnectionConfig(options) {
   }
 
   this.maxPacketSize = 0;
-  this.charsetNumber = (options.charset)
-    ? ConnectionConfig.getCharsetNumber(options.charset)
-    : options.charsetNumber || Charsets.UTF8_GENERAL_CI;
+  this.charsetNumber = (options.charset) ? ConnectionConfig.getCharsetNumber(options.charset) : options.charsetNumber || Charsets.UTF8_GENERAL_CI;
 
   // Set the client flags
-  var defaultFlags = ConnectionConfig.getDefaultFlags(options);
-  this.clientFlags = ConnectionConfig.mergeFlags(defaultFlags, options.flags);
+  this.clientFlags = ConnectionConfig._getClientFlags(this.multipleStatements, this.flags);
 }
 
+var _clientFlagsCache = {};
+
+ConnectionConfig._getClientFlags = function(multipleStatements, flags) {
+  var key = flags + multipleStatements;
+
+  if (_clientFlagsCache[key] === undefined) {
+    var defaultFlags = ConnectionConfig.getDefaultFlags(multipleStatements);
+    _clientFlagsCache[key] = ConnectionConfig.mergeFlags(defaultFlags, flags);
+  }
+
+  return _clientFlagsCache[key];
+};
+
 ConnectionConfig.mergeFlags = function mergeFlags(defaultFlags, userFlags) {
   var allFlags = ConnectionConfig.parseFlagList(defaultFlags);
   var newFlags = ConnectionConfig.parseFlagList(userFlags);
 
   // Merge the new flags
-  for (var flag in newFlags) {
-    if (allFlags[flag] !== false) {
-      allFlags[flag] = newFlags[flag];
+  for (var newFlagKey in newFlags) {
```

Comment:
Why rename this variable? Seems unrelated to the pull request. Please undo or explain how it fits into changing the pool.